tutorials/wmt/README.md

```diff
@@ -0,0 +1,251 @@
+python -m apply_bpe -c bpe.codes --vocabulary bpe.vocab.en --vocabulary-threshold 50 < newstest2016.tc.en > newstest2016.tc.BPE.en
+```
+
+Looking at the data you can see how words are split into subwords separated by the special sequence `@@`:
+```
+Globaldarlehen sind Kreditlinien an zwischengeschaltete Institute -> Glob@@ al@@ dar@@ lehen sind Kredit@@ linien an zwischen@@ gesch@@ al@@ tete Institute
+```
+
+## Training
+
+Having preprocessed our data we can start training.
+Note that Sockeye will load all training data into memory in order to be able to easily reshuffle after every epoch.
+Depending on the amount of RAM you have available you might want to reduce size of the training corpus for this tutorial:
+```bash
+# (Optional: run this if you have limited RAM on the training machine) 
+head -n 200000 corpus.tc.BPE.de > corpus.tc.BPE.de.tmp && mv corpus.tc.BPE.de.tmp corpus.tc.BPE.de
+head -n 200000 corpus.tc.BPE.en > corpus.tc.BPE.en.tmp && mv corpus.tc.BPE.en.tmp corpus.tc.BPE.en
+```
+That said, we can how kick off the training process:
+```bash
+python -m sockeye.train -s corpus.tc.BPE.de \
+                        -t corpus.tc.BPE.en \
+                        -vs newstest2016.tc.BPE.de \
+                        -vt newstest2016.tc.BPE.en \
+                        --num-embed 256 \
+                        --rnn-num-hidden 512 \
+                        --attention-type dot \
+                        --max-seq-len 60 \
+                        --monitor-bleu 500 \
+                        --use-tensorboard \
+                        --use-cpu \
+                        -o wmt_model
+```
+
+This will train a 1-layer bi-LSTM encoder, 1 layer LSTM decoder with dot attention.
+Sockeye offers a whole variety of different options regarding the model architecture,
+such as stacked RNNs with residual connections (`--num-layers`, `--rnn-residual-connections`),
+[Transformer](https://arxiv.org/abs/1706.03762) encoder and decoder (`--encoder transformer`, `--decoder transformer`),
+various RNN (`--rnn-cell-type`) and attention (`--attention-type`) types and more.  
+
+There are also several parameters controlling training itself.
+Unless you specify a different optimizer (`--optimizer`) [Adam](https://arxiv.org/abs/1412.6980) will be used.
+Additionally, you can control the batch size (`--batch-size`), the learning rate schedule (`--learning-rate-schedule`)
+and other parameters relevant for training.
+
+Training will run until the validation perplexity stops improving.
+Alternatively, you can track BLEU on the validation set (`--optimized-metric bleu`).
+Sockeye will then at every checkpoint start a decoder in a separate process running on the same device as training.
+To make sure the decoder finishes before the next checkpoint one can subsample the validation set for
+BLEU score calculation.
+For example `--monitor-bleu 500` will calculate BLEU on a random subset of 500 sentences.
```

Comment:
maybe note that this random set is sampled once and based on a fixed random seed

Reply:
done